rootfs/etc/nginx/lua/balancer/sticky.lua

```diff
@@ -43,6 +44,20 @@ function _M.set_cookie(self, value)
     cookie_path = ngx.var.location_path
   end
 
+  local cookie_samesite = self.cookie_session_affinity.samesite
+  if cookie_samesite then
+    local cookie_conditional_samesite_none = self.cookie_session_affinity.conditional_samesite_none
+    if cookie_conditional_samesite_none
+        and cookie_samesite == "None"
+        and not same_site.same_site_none_compatible(ngx.var.http_user_agent) then
+      cookie_samesite = nil
+    end
+  end
+
+  if cookie_samesite then
+    cookie_path = cookie_path .. "; SameSite=" .. cookie_samesite
+  end
```

Comment:
The lua-resty-cookie library has native support for `samesite`, please use that attribute instead of misusing `path` attribute. All you need to do is: `samesite = cookie_samesite,` below after `secure = ngx.var.https == "on",`.

Reply:
Would it be better to add a TODO here to fix this after that library adds SameSite=None support?

Reply:
FYI the upstream change was merged a day or two ago.

Reply:
@RulerOf yes, and the nginx base image already contains such version https://github.com/kubernetes/ingress-nginx/pull/5011
Now, someone needs to update the lua code :)

Pull requests are welcome

Reply:
Will do both of these this weekend!

Reply:
Drafted, will poke on tests later on this weekend. https://github.com/kubernetes/ingress-nginx/pull/5040